docs/kubernetes_setup.md

```diff
@@ -0,0 +1,188 @@
+# Setting up apiserver-network-proxy in Kubernetes
+
+This example will set up apiserver-network-proxy server as a Deployment in the master and apiserver-network-proxy agent as a DaemontSet on the worker nodes.
+
+## Prerequisites
+
+In order to set up the network-proxy, it's needed to generate 2 CA. One for the master communication, `konnectivity-server-ca.crt`
+and one for the cluster communication, `konnectivity-cluster-ca.crt`.
+
+Then, create:
+- A client authentication certificate for the network-proxy agent, `konnectivity-agent-client.crt` and `konnectivity-agent-client.key`.
+- A server authentication certificate for the cluster side of the network-proxy server, `konnectivity-cluster.crt` and `konnectivity-cluster.key` (one for each node or one per node).
+- A client authentication certificate for the Kubernetes API server, `konnectivity-server-client.crt` and `konnectivity-server-client.key`.
+- A server authentication certificate for the server side of the network-proxy server, `konnectivity-server.crt` and `konnectivity-server.key`.
+
```

Comment:
It'd be nice/less confusing if the names here matched what the repo's command `make certs` output, or if the make target was updated to reflect these names. Personally I prefer the names you have here. You could modify the `certs` make target in another PR, but if you did it in this one then you could include running that command in the instructions such that you're providing a full "out-of-box" guide to get up and running. The current make target outputs a directory at the root of the repo which looks like this:

```
certs
├── agent
│   ├── issued
│   │   ├── ca.crt
│   │   ├── proxy-agent.crt
│   │   └── proxy-master.crt
│   └── private
│       ├── ca.key
│       ├── proxy-agent.key
│       └── proxy-master.key
└── master
    ├── issued
    │   ├── ca.crt
    │   ├── proxy-client.crt
    │   └── proxy-master.crt
    └── private
        ├── ca.key
        ├── proxy-client.key
        └── proxy-master.key
```

Reply:
Yep I agree. I'll think about ut

Reply:
+1